grammar.js

```diff
@@ -1341,292 +1341,280 @@ module.exports = grammar({
+          PREC.control,
+          seq(
+            "for",
+            field(
+              "enumerators",
+              choice(
+                seq("(", $.enumerators, ")"),
+                seq("{", $.enumerators, "}"),
+              ),
+            ),
+            choice(
+              seq(field("body", $.expression)),
+              seq("yield", field("body", $._indentable_expression)),
+            ),
+          ),
         ),
-      )),
-      prec.right(PREC.control, seq(
-        'for',
-        field('enumerators', $.enumerators),
-        choice(
-          seq('do', field('body', $._indentable_expression)),
-          seq('yield', field('body', $._indentable_expression)),
+        prec.right(
+          PREC.control,
+          seq(
+            "for",
+            field("enumerators", $.enumerators),
+            choice(
+              seq("do", field("body", $._indentable_expression)),
+              seq("yield", field("body", $._indentable_expression)),
+            ),
+          ),
         ),
-      )),
-    ),
-
-    enumerators: $ => choice(
-      seq(
-        sep1($._semicolon, $.enumerator),
-        optional($._automatic_semicolon)
       ),
-      seq(
-        $._indent,
-        sep1($._semicolon, $.enumerator),
-        optional($._automatic_semicolon),
-        $._outdent,
+
+    enumerators: $ =>
+      choice(
+        seq(sep1($._semicolon, $.enumerator), optional($._automatic_semicolon)),
+        seq(
+          $._indent,
+          sep1($._semicolon, $.enumerator),
+          optional($._automatic_semicolon),
+          $._outdent,
+        ),
       ),
-    ),
 
     /**
      *   Enumerator        ::=  Generator
      *                       |  Guard {Guard}
      *                       |  Pattern1 '=' Expr
      */
-    enumerator: $ => choice(
-      seq(
-        optional('case'),
-        $._pattern,
-        choice('<-', '='),
-        $.expression,
-        optional($.guard)
+    enumerator: $ =>
+      choice(
+        seq(
+          optional("case"),
+          $._pattern,
+          choice("<-", "="),
+          $.expression,
+          optional($.guard),
+        ),
+        repeat1($.guard),
       ),
-      repeat1($.guard),
-    ),
 
-    comment: $ => seq(token('//'), choice(
-      $.using_directive,
-      $._comment_text,
-    )),
+    comment: $ => seq(token("//"), choice($.using_directive, $._comment_text)),
 
     _comment_text: $ => token(prec(PREC.comment, /.*/)),
 
-    using_directive: $ => seq(
-      token.immediate(prec(PREC.using_directive, '>')),
-      token('using'),
-      $.using_directive_key,
-      $.using_directive_value,
-    ),
+    using_directive: $ =>
+      seq(
+        token.immediate(prec(PREC.using_directive, ">")),
+        token("using"),
+        $.using_directive_key,
+        $.using_directive_value,
+      ),
     using_directive_key: $ => token(/[^\s]+/),
     using_directive_value: $ => token(/.*/),
 
-    block_comment: $ => seq(
```

Comment:
In such cases, multi-line formatting is intentional and (very subjectively) more easily comprehended. I'd suggest disable auto-wrapping by setting `printWidth` to null

Reply:
Yea. Unfortunately, I don't think there's a way to preserve end-of-line.
It's either going to make all the lines long, or always add EOL.

Reply:
Ah, okay